spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -473,6 +473,22 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
     this
   }
 
+  /**
+   * Save this module with class tag to path with protobuf format
+   * @param path path to save module, local file system, HDFS and Amazon S3 is supported.
+   *             HDFS path should be like "hdfs://[host]:[port]/xxx"
+   *             Amazon S3 path should be like "s3a://bucket/xxx"
+   * @param moduleTag  module tag
+   * @param overWrite if overwrite
+   * @return self
+   */
+  def saveModuleWithTag(path : String, moduleTag : ModuleTag, overWrite
```

Comment:
Why do we need a new API? Just make moduleTag an optional argument in saveModule